[11.0 P6] Blazor Preview 6 coverage (#37322)
dotnet/AspNetCore.Docs: aspnetcore/blazor/components/virtualization.md

Reviewer: ```suggestion
    <div data-blazor-virtualize-reserved-height="3400" aria-hidden="true"></div>
```
With a monkier.

Reviewer: ```suggestion
    <<div data-blazor-virtualize-reserved-height="1100" aria-hidden="true"></div>
```
With a monkier.

* <xref:blazor/components/control-head-content>

:::moniker-end

:::moniker range=">= aspnetcore-7.0"

## Content Security Policy (CSP) compliance

The `Virtualize` component renders dynamic inline `style` attributes on its spacer and placeholder elements because spacer heights are calculated at runtime based on scroll position, item count, and average item size, which change on every scroll interaction.

> To avoid CSP violations, `Virtualize` components:

Reviewer: This is a bit too much implementation details. I will try to come up with a shorter, less detailed update. Something like:

"Spacer heights are set via the browser's CSS Object Model from the `data-blazor-virtualize-reserved-height` attribute value, rather than as inline style attributes. This makes the component compatible with strict [Content Security Policy (CSP)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Guides/CSP) configurations."

 would be enough. Without the explanation how we render heights dynamically and that we use `MutationObserver`. Together with the update in lines 370-376, the developer would have all the important info.

> * Render CSS styles in a `data-blazor-style` attribute instead of a `style` attribute.

Reviewer: We extracted the only variable from the style that has to be computed on the server and that's the height. The rest of styles is applied on the client. That's why it used to be styles, now it's just `data-blazor-virtualize-reserved-height`.

```suggestion
* Render CSS height in a `data-blazor-virtualize-reserved-height` attribute instead of a `style` attribute.
```
